tls/s2n_record_write.c

```diff
@@ -263,7 +278,11 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
     /* Start the MAC with the sequence number */
     POSIX_GUARD(s2n_hmac_update(mac, sequence_number, S2N_TLS_SEQUENCE_NUM_LEN));
 
-    POSIX_GUARD(s2n_stuffer_resize_if_empty(&conn->out, S2N_LARGE_RECORD_LENGTH));
+    if (s2n_stuffer_is_freed(&conn->out)) {
+        uint16_t max_wire_record_size = 0;
+        POSIX_GUARD_RESULT(s2n_record_max_write_size(conn, max_write_payload_size, &max_wire_record_size));
```

Comment:
a doc nit: this made more sense to me when i realized that this could be called on the server side after MFL negotiation happens since the server hasn't done any writes yet. It would be useful to document the cases of the "source" of max_write_payload_size here:

1. max record size set by s2n_connection_prefer_throughput(), s2n_connection_prefer_low_latency()
2. When a server, the negotiated MFL, if it's lower than the value set in 1).

It would also be useful to note that this condition won't be hit again during the lifetime of a connection unless an application calls s2n_connection_release_buffers